DataStructures/SortedList/SortedList.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text;
+
+using Utilities.Exceptions;
+
+namespace DataStructures.SortedList
+{
+    /// <summary>
+    /// Implementation of a Sorted List. It uses Quick Sort Algorithm.
+    /// </summary>
+    /// <typeparam name="T">Generic Type.</typeparam>
+    public class SortedList<T> : IComparisonSorter<T>, IEnumerable<T>
+    {
+        private readonly List<T> list;
+
+        public SortedList() => list = new List<T>();
+
+        public SortedList(List<T> myList)
+        {
+            list = new List<T>(myList.Count);
+            list = myList;
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return list.GetEnumerator();
+        }
+
+        public IEnumerator<T> GetEnumerator()
+        {
+            return ((IEnumerable<T>)list).GetEnumerator();
+        }
+
+        /// <summary>
+        /// Insert item in a sorted list.
+        /// </summary>
+        /// <param name="item">Item of Type T to be inserted in the list.</param>
+        public void Insert(T item)
+        {
+            // Use BinarySearch() method to determine if the item  exists in the list
+            int binarysearchIndex = list.BinarySearch(item);
+
+            if (binarysearchIndex < 0)
+            {
+                // Bitwise complement on the returned negative integer, gives positive integer.
+                list.Insert(~binarysearchIndex, item);
+            }
+            else
+            {
+                list.Insert(binarysearchIndex, item);
+            }
+
+            var intComparer = new IntComparer();
+            list.Sort(0, list.Count, (IComparer<T>)intComparer);
+        }
+
+        /// <summary>
+        /// Remove item from a sorted list.
+        /// </summary>
+        /// <param name="item">Item of Type T to be removed from the list.</param>
+        public void Remove(T item)
+        {
+            int binarysearchIndex = list.BinarySearch(item);
+
+            if (binarysearchIndex > 0)
+            {
+                list.RemoveAt(binarysearchIndex);
+            }
+            else
+            {
+                throw new ArgumentOutOfRangeException("Item is not present in the list");
```

Comment:
Codacy found an issue: [The parameter name 'Item is not present in the list' is not declared in the argument list.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6375010)